contracts/0.8.9/WithdrawalQueueERC721.sol

```diff
@@ -338,4 +349,31 @@ contract WithdrawalQueueERC721 is IERC721Metadata, WithdrawalQueue {
             baseURI.slot := position
         }
     }
+
+    function _constructTokenUri(uint256 _requestId) internal view returns (string memory) {
+        string memory baseURI = _getBaseURI().value;
+        if (bytes(baseURI).length == 0) return "";
+
+        // ${baseUri}/${_requestId}?state=finalized|unfinalized&amount=${amount}&created_at=${timestamp}
+        // we still have no string.concat in 0.8.9, so we have to do it with bytes
+        bool finalized = _requestId <= getLastFinalizedRequestId();
+        return string(
+            bytes.concat(
+                bytes(baseURI),
+                bytes(_requestId.toString()),
+                bytes("?status="),
+                bytes(finalized ? "finalized" : "pending"),
+                bytes("&amount="),
+                bytes(
```

Comment:
Let's add both amounts unconditionally.

It's vital to have the requested `stETH` unchanged.
While claimable changes upon the finalization.